alphagov/multicloud-deploy
# [#103903920] Cloud Foundry phoenix jobs

# What

We are now saving time by reusing our environments, but we still need to test building an environment from scratch. This PR adds Ansible templates to generate the jobs in Jenkins.

We manage passwords with `paas-pass` so this had to be installed on Jenkins. And we need SSH keys to access the credentials repository so the management of these keys has been added.

Terraform has been upgraded to 0.6.3 because the syntax in CF terraform code requires it.

This also includes some optimisations to save time building and running Jenkins.

Related PRs:
- https://github.com/alphagov/cf-terraform/pull/47
- https://github.gds/multicloudpaas/credentials/pull/18
# How to review
- Ideally merge the first 2 PRs
- Run `vagrant up`
- Open jenkins on https://127.0.0.1:8443
- Start job `ci-cf-deploy-phoenix-aws` or `ci-cf-deploy-phoenix-gce`
- It should build a new environment, run smoke tests on it, and destroy it
- Wait until 7am until it starts automatically
# Who can review

Anyone but @saliceti, @jimconner , @Jonty
